sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -372,6 +425,31 @@ protected static SqlParts selectStatementForAndClause(List<Object> clause,
     }
 
     @SuppressWarnings("unchecked")
+    protected static SqlParts selectStatementForTextClause(Object clause,
+                                                           String indexName) {
+        if (clause == null) {
+            return null;  // no query here
+        }
+
+        if (indexName == null || indexName.isEmpty()) {
+            return null;
+        }
+
+        if (!(clause instanceof Map)) {
+            return null;  // should never get here as this would not pass normalization
+        }
+
+        Map<String, Object> textClause = (Map<String, Object>) clause;
+        Map<String, String> searchClause = (Map<String, String>) textClause.get(TEXT);
+
+        String tableName = IndexManager.tableNameForIndex(indexName);
+        String search = searchClause.get(SEARCH);
+
+        String sql = String.format("SELECT _id FROM %s WHERE %s MATCH ?", tableName, tableName);
+        return SqlParts.partsForSql(sql, new String[]{ String.format("'%s'", search) });
```

Comment:
Is this placeholder correct, looks like if my search includes a `'` I can SQL inject here?